### PR TITLE
add a missing include for std::accumulate()

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

**Comment by drwells** [comment]
Nice catch. I thought this was in `algorithm`.

Out of curiosity, does the current version compile correctly on your machine without `#include <numeric>`? I was able to compile (and run tests) without errors when I approved #2214.

Would you mind taking out the `#include <algorithm>`? That was only added in the patch that added `std::accumulate`.

File: include/deal.II/grid/tria.h [hunk 1]
```diff
@@ -40,6 +40,7 @@ DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
 #include <list>
 #include <map>
 #include <algorithm>
+#include <numeric> // need for std::accumulate
```
Comment (bangerth) [inline include/deal.II/grid/tria.h]: Ah, I just added <algorithm> yesterday. Please remove <algorithm> again, and then also drop the comment.

**Comment by davydden** [comment]
@drwells it did not compile for me with `#include <algorithm>` with clang.

**Comment by davydden** [comment]
could we merge that as it does not build otherwise.

**Pull request merged**